src/server/server.ts

```diff
@@ -1,127 +1,24 @@
-import { ServerParsedArgs, serverOptions } from 'vs/server/node/serverEnvironmentService'
-import product from 'vs/platform/product/common/product'
-import { AddressInfo, ListenOptions, Socket } from 'net'
-import http from 'http'
+import { IProductService } from '../services'
 
-export async function start (options: ListenOptions): Promise<void> {
-  let _remoteExtensionHostAgentServer: IServerAPI | null = null
-  let _remoteExtensionHostAgentServerPromise: Promise<IServerAPI> | null = null
-  const getRemoteExtensionHostAgentServer = (): Promise<IServerAPI> => {
-    if (_remoteExtensionHostAgentServerPromise == null) {
-      _remoteExtensionHostAgentServerPromise = createServer(address).then(server => {
-        _remoteExtensionHostAgentServer = server
-        return server
-      })
-    }
-    return _remoteExtensionHostAgentServerPromise
-  }
-
-  let address: AddressInfo | string | null = null
-  const server = http.createServer(async (req, res) => {
-    const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer()
-    return remoteExtensionHostAgentServer.handleRequest(req, res)
-  })
-  server.on('upgrade', async (req, socket) => {
-    const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer()
-    return remoteExtensionHostAgentServer.handleUpgrade(req, <Socket>socket)
-  })
-  server.on('error', async (err) => {
-    const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer()
-    return remoteExtensionHostAgentServer.handleServerError(err)
-  })
-
-  server.listen(options, async () => {
-    address = server.address()
-    if (address === null) {
-      throw new Error('Unexpected server address')
-    }
-
-    // eslint-disable-next-line no-console
-    console.log(`Server bound to ${typeof address === 'string' ? address : `${address.address}:${address.port} (${address.family})`}`)
-
-    await getRemoteExtensionHostAgentServer()
-  })
-
-  process.on('exit', () => {
-    server.close()
-    if (_remoteExtensionHostAgentServer != null) {
-      _remoteExtensionHostAgentServer.dispose()
-    }
-  })
+const thisWithVSCodeParams = globalThis as typeof globalThis & {
+  _VSCODE_PRODUCT_JSON: Partial<IProductService>
+  _VSCODE_PACKAGE_JSON: { version: string }
 }
 
-function parseRange (strRange: string) {
-  const match = strRange.match(/^(\d+)-(\d+)$/)
-  if (match != null) {
-    const start = parseInt(match[1]!, 10); const end = parseInt(match[2]!, 10)
-    if (start > 0 && start <= end && end <= 65535) {
-      return { start, end }
-    }
+// Initialize the product information for the server, including the extension gallery URL.
+thisWithVSCodeParams._VSCODE_PRODUCT_JSON = {
+  extensionsGallery: {
+    serviceUrl: 'https://open-vsx.org/vscode/gallery',
+    itemUrl: 'https://open-vsx.org/vscode/item',
+    resourceUrlTemplate: 'https://open-vsx.org/vscode/unpkg/{publisher}/{name}/{version}/{path}',
```

Comment:
So we can't override it on the server?

Reply:
No, it's loaded as a side effect so this code has to load first.

Reply:
In VSCode it's injected during build.

Reply:
Can't we allow to override it using a cli parameter or something?

Reply:
What about an env variable which contains an optional product.json file to load then?

Reply:
Oh huh, interesting... I thought I had built that. I think I deleted a commit in the history because that's what I exactly did! I will update the PR.

Reply:
Updated the PR with custom env var for giving product.json